### DISCUSSION TITLE
Lull skills discussion

### DISCUSSION THREAD

**Discussion by JesterWizard** [MokhaLeee/fe8u-cskillsys-kernel#264] [General]
So the lull skills work in the ASM system by looking for a specific bit set in the unit's state debuff table and then negating the boost. However, we can't really do that here can we? So how do you propose we implement these skills?

My thoughts involved one of the following:

- Apply the lull skill if the enemy unit has a corresponding rally skill (so apply Lull Def if the enemy has Rally Def)
- Apply the lull skill if the enemy unit has a boost in that stat (so if the enemy has a +4 boost in DEF, then apply Lull Def)

**Comment by MokhaLeee** [comment]
It need more bisc works to add new API for debuffs rather than directly check on stat-debuff index.

Previously in order to enable Clearance and LightAndDark skill we have done in some process, https://github.com/MokhaLeee/fe8u-cskillsys-kernel/commit/471def9ce. But it seems that it need to more on it. You can pull the skill info first.

**Reply by JesterWizard** [reply]
Since the current implementation of buffs/debuffs obscures what was actually applied to save RAM space, I've made a working version of this on my fork by choosing to make it an aura skill that blocks the application of all rallies in a 3 tile range of the user.
